redis-persistence/src/main/java/com/netflix/conductor/dao/DynoQueueDAOProvider.java

```diff
@@ -0,0 +1,34 @@
+package com.netflix.conductor.dao;
+
+
+import com.netflix.conductor.core.config.Configuration;
+import com.netflix.conductor.dao.dynomite.queue.DynoQueueDAO;
+import com.netflix.dyno.connectionpool.HostSupplier;
+import com.netflix.dyno.queues.redis.DynoShardSupplier;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import redis.clients.jedis.JedisCommands;
+
```

Comment:
@Singleton?

Reply:
There is an interesting question whether that makes the provider a single or the result. I am not super clear reading the Guice docs how that will play out.  I *think* it is better to mark the class the is being provided as a singleton, if at all possible.  I'll look into this some more though.

Reply:
The current bootstrap code seems to totally dodge this constructor 
```
@Inject
public DynoQueueDAO(DiscoveryClient dc, Configuration config) {
```
Which seems sub-optimal.  Do you know if there is a reason for that?

In the short term I can make the Provider itself a singleton and then ensure there is only one instance within it.  But that seems like more of a hack that could end badly.